src/cqueues.c

```diff
@@ -1982,10 +1982,22 @@ static cqs_status_t cqueue_resume(lua_State *L, struct cqueue *Q, struct callinf
 		}
 	} else {
 		nargs = lua_gettop(T->L);
-		if (status != LUA_YIELD) {
+		if (status == LUA_OK && lua_getstack(T->L, 0, &(lua_Debug){}) > 0) {
```

Comment:
This little bit was adapted from the `coroutine.status` implementation: http://www.lua.org/source/5.3/lcorolib.c.html#luaB_costatus